chroma-agent/chroma_agent/agent_daemon.py

```diff
@@ -50,112 +40,33 @@ def boot_time(self):
                 return datetime.datetime.fromtimestamp(int(val))
 
 
-def kill_orphans_of(parent_pid, timeout = 5):
-    """
-    If a previous instance of the service left its PID file behind (unclean stop) then
-    we will see if there are any subprocesses hanging around, and kill them.  This is done
-    to avoid e.g. a mkfs from a previous agent instance still being in progress when we
-    start another one (imagine restarting chroma-agent during a long format and then clicking
-    the format button again in the GUI).
-
-    :param parent_pid: Find orphans from this PID
-    :param timeout: Wait this long for orphan processes to respond to SIGKILL
-    :return: None
-    """
-    victims = []
-    for pid in [int(pid) for pid in os.listdir('/proc') if pid.isdigit()]:
-        try:
-            stat = open("/proc/%s/stat" % pid).read().strip()
-            pgid = int(stat.split()[4])
-            ppid = int(stat.split()[3])
-        except OSError, e:
-            if e.errno != errno.ENOENT:
-                raise
-        else:
-            if int(pgid) == int(parent_pid) and ppid == 1:
-                sys.stderr.write("Killing orphan process %s from previous instance %s\n" % (pid, parent_pid))
-                victims.append(pid)
-                os.kill(pid, signal.SIGKILL)
-
-    n = 0
-    while True:
-        for pid in victims:
-            if not os.path.exists("/proc/%s" % pid):
-                sys.stderr.write("Process %s terminated successfully\n")
-                victims.remove(pid)
-
-        if victims:
-            n += 1
-
-            if n > timeout:
-                raise RuntimeError("Failed to kill orphan processes %s after %s seconds" % (victims, timeout))
-
-            time.sleep(1)
-        else:
-            break
-
-
 def main():
-    """Daemonize and handle unexpected exceptions"""
-    parser = argparse.ArgumentParser(description="Intel® Manager for Lustre* software Agent")
-    parser.add_argument("--foreground", action="store_true")
+    """handle unexpected exceptions"""
+    parser = argparse.ArgumentParser(
+        description="Intel® Manager for Lustre* software Agent")
+
     parser.add_argument("--publish-zconf", action="store_true")
-    parser.add_argument("--pid-file", default = "/var/run/chroma-agent.pid")
     args = parser.parse_args()
 
     # FIXME: at startup, if there is a PID file hanging around, find any
```

Comment:
we may not need this comment anymore, I would have thought that with systemd managing the service we shouldn't end up with leftover child PIDs

Reply:
Yeah, I think so, I'll remove.